base/iobuffer.jl

```diff
@@ -390,18 +397,28 @@ end
 function take!(io::IOBuffer)
     ismarked(io) && unmark(io)
     if io.seekable
-        data = io.data
         if io.writable
-            maxsize = (io.maxsize == typemax(Int) ? 0 : min(length(io.data),io.maxsize))
```

Comment:
Note the new version loses the sizehint that we used to get from `length(io.data)` for re-using the same `IOBuffer` object, but this was only for finite `io.maxsize` and does not seem to a common pattern?

Alternatively, maybe we should save the `sizehint`, and set it to `max(io.sizehint, length(io.data))` on a `take!`?